### PR TITLE
Fixes the issue On V6, for android tests, where the incorrect deviceName is displayed in allure reports

### PR DESCRIPTION
**Pull request by jags14385**
## Proposed changes

Fixes the issue On V6, for android tests, where the incorrect deviceName is displayed in allure reports #5489 

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ x] I have added tests that prove my fix is effective or that my feature works
- [ x] I have added necessary documentation (if appropriate)
- [ x] I have added proper type definitions for new commands (if appropriate)

## Further comments

In the fix, desired.PlatformVersion takes precedence over platformVersion(as shown in the screenshot).

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

File: packages/wdio-allure-reporter/src/index.js [hunk 1]
```diff
@@ -110,8 +110,11 @@ class AllureReporter extends WDIOReporter {
         const currentTest = this.allure.getCurrentTest()
 
         if (!this.isMultiremote) {
-            const { browserName, deviceName } = this.capabilities
-            const targetName = browserName || deviceName || cid
+            const { browserName, deviceName, desired } = this.capabilities
+            let targetName = browserName || deviceName || cid
+            if(desired && desired.deviceName && desired.platformVersion){
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/index.js:115]: minor formatting, also let's add a comment
```suggestion
            /**
             * custom mobile grids can have device information in a `desird` cap
             */
            if (desired && desired.deviceName && desired.platformVersion) {
```

File: packages/wdio-allure-reporter/tests/allureFeatures.test.js [hunk 2]
```diff
@@ -377,6 +377,29 @@ describe('auxiliary methods', () => {
...
+                platformName: 'Android',
+                automationName: 'UiAutomator2',
+                deviceName: 'Android GoogleAPI Emulator',
+                platformVersion: '6.0',
+                noReset: true,
+            }
+        }
+        const reporter = new AllureReporter({ stdout: true })
+        let addParameter = jest.fn()
+        let currentTestMock = { addParameter, addLabel: jest.fn() }
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/tests/allureFeatures.test.js:394]: no need to define it in an extra variable, also we should use `const` if we decide not to change the variable value
```suggestion
        const currentTestMock = {
            addParameter: jest.fn(),
            addLabel: jest.fn()
        }
```

File: packages/wdio-allure-reporter/src/index.js [hunk 3]
```diff
@@ -110,8 +110,11 @@ class AllureReporter extends WDIOReporter {
...
+                targetName = desired.deviceName+'-'+desired.platformVersion
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/src/index.js:116]: How about we separate by space
```suggestion
                targetName = `${desired.deviceName} ${desired.platformVersion}`
```

File: packages/wdio-allure-reporter/tests/allureFeatures.test.js [hunk 4]
```diff
@@ -377,6 +377,29 @@ describe('auxiliary methods', () => {
...
+        reporter.allure.getCurrentTest = jest.fn().mockReturnValue(currentTestMock)
+        reporter.allure.startCase = jest.fn()
+        reporter.isMultiRemote = false
+        reporter.capabilities = capabilities
+        reporter.onTestStart({ cid: '0-0', title: 'SomeTest' })
+        expect(reporter.allure.getCurrentTest.mock.calls.length).toBe(1)
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/tests/allureFeatures.test.js:400]:
```suggestion
        expect(reporter.allure.getCurrentTest).toBeCalledTimes(1)
```

File: packages/wdio-allure-reporter/tests/allureFeatures.test.js [hunk 5]
```diff
@@ -377,6 +377,29 @@ describe('auxiliary methods', () => {
...
+        expect(addParameter).toHaveBeenCalledWith('argument', 'device', 'Android GoogleAPI Emulator-6.0')
```
Comment (christian-bromann) [inline packages/wdio-allure-reporter/tests/allureFeatures.test.js:401]:
```suggestion
        expect(currentTestMock.addParameter).toHaveBeenCalledWith('argument', 'device', 'Android GoogleAPI Emulator-6.0')
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Thanks for tackling this, here are some suggestions:

**Comment by jags14385** [comment]
Thanks for helping me out too :) . I have made changes as per the feedback.

**Review by christian-bromann — APPROVED** [review]
👍 thanks! And congrats on your first contribution 👏

**Pull request merged**